ostd/src/arch/x86/trap/mod.rs

```diff
@@ -220,7 +221,17 @@ pub fn is_kernel_interrupted() -> bool {
 /// Handle traps (only from kernel).
 #[no_mangle]
 extern "sysv64" fn trap_handler(f: &mut TrapFrame) {
-    match CpuException::to_cpu_exception(f.trap_num as u16) {
+    /// Ref: https://wiki.osdev.org/CPU_Registers_x86#EFLAGS_Register
+    const IRQ_ENABLED_MASK: usize = 1 << 9;
+
+    let cpu_exception = CpuException::to_cpu_exception(f.trap_num as u16);
+    // Ensures that the IRQ state during the trap that handles exceptions remains
+    // consistent with the state before entering.
+    let is_trap_irq_enabled = (f.rflags & IRQ_ENABLED_MASK > 0) && cpu_exception.is_some();
+    if is_trap_irq_enabled {
+        enable_local();
+    }
```

Comment:
I think isolating `enable_local`/`disable_local` into separate code blocks improves the readability.

```rust
    let cpu_exception = CpuException::to_cpu_exception(f.trap_num as u16);

    // Ensures that the IRQ state during the trap that handles exceptions remains
    // consistent with the state before entering.
    let is_trap_irq_enabled = cpu_exception.is_some()
        && (f.rflags & x86_64::registers::rflags::RFlags::INTERRUPT_FLAG.bits() > 0);
    if is_trap_irq_enabled {
        enable_local();
    }

    match cpu_exception {
        // ...
    }

    if is_trap_irq_enabled {
        disable_local();
    }
```